src/components/publish-site-button.tsx

```diff
@@ -27,8 +27,12 @@ export const PublishSiteButton = () => {
 
 	const handlePublishClick = useCallback( () => {
 		setSelectedTab( 'sync' );
-		dispatch( connectedSitesActions.openModal( 'push' ) );
-	}, [ setSelectedTab, dispatch ] );
+		if ( ! user ) {
+			authenticate();
+		} else {
+			dispatch( connectedSitesActions.openModal( 'push' ) );
```

Comment:
We can consider dispatching the action of open modal in any case. Outside of the else. That way the modal will open once the user is logged in.

Something like:

```
if ( ! user ) {
   authenticate();
}
dispatch( connectedSitesActions.openModal( 'push' ) );
```

Reply:
I have tested clicking multiple times, and it opens multiple browsers similarly to clicking multiple times on `Log in to WordPress.com`, which I think is fine.

Also, when clicking multiple sites, _the last one wins_ which in my opinion is also correct.

I have pushed the change as part of 579661ee, please let me know if you find any issues